src/compiler.ts

```diff
@@ -0,0 +1,157 @@
+import { filter, keys, clone, forEach, flow, negate, eq, includes, gte, gt, lte, lt, get, map, some, flatten, isString, isRegExp } from 'lodash/fp';
+import hi from './hidash';
+import { Query } from './types';
+
+export default class QueryCompiler {
+  compile(query: Query) {
+    const filters = this._compilePredicates(query) as any[];
+    let filtered;
+    if (filters.length === 1) {
+      filtered = filter(filters[0]);
+    } else {
+      filtered = filters.map(function (v) {
+        return filter(v);
+      });
+      filtered = flow(filtered);
+    }
+    return filtered;
+  }
+
+  _compilePredicates(query: Query): any {
```

Comment:
Not sure what the return type should be, if I remove the any it complains and if I add right types it still complains.